### PR TITLE
Added TypeScript typings and updated README

### PR DESCRIPTION
**Pull request by slavikdenis**
- added typings, completed the https://github.com/Richou/react-native-android-location-enabler/pull/22 PR
- added missing error code to README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,35 +27,42 @@ Since ReactNative 0.60.0 and ReactNative Cli 2.0.0, no action is needed to add a
...
+	})
+	.catch((err) => {
+		// The user has not accepted to enable the location services or something went wrong during the process
+		// "err" : { "code" : "ERR00|ERR01|ERR02|ERR03", "message" : "message"}
+		// codes :
+		//  - ERR00 : The user has clicked on Cancel button in the popup
+		//  - ERR01 : If the Settings change are unavailable
+		//  - ERR02 : If the popup has failed to open
+		//  - ERR03 : Internal error
+	});
```
Comment (Richou) [inline README.md:67]: Please use 2 spaces for indent, instead of tab, thx.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+interface PromptFuncOptions {
```
Comment (Richou) [inline index.d.ts:1]: Please use 2 spaces for indent, instead of tab, thx.

**Comment by slavikdenis** [comment]
@Richou added wanter changes :)

**Comment by Richou** [comment]
@slavikdenis Thank you for the PR, I will release a new version asap.

**Comment by MBach** [comment]
Hi @Richou 
Do you have an opportunity to make a new release soon 👍🏻 ?

**Comment by Richou** [comment]
Hi @MBach the new version 1.2.2 has been released.
Please let me know you have problems with it.

Regards.

**Comment by MBach** [comment]
Seems to work for typings, but if I refuse, I have an Unhandled Promise Rejection!
```javascript
  const activateLocation = async () => {
    const res: PromptFuncResponse = await RNAndroidLocationEnabler.promptForEnableLocationIfNeeded({
      interval: 10000,
      fastInterval: 5000
    })
    console.log('promptForEnableLocationIfNeeded', res)
    switch (res) {
      case 'already-enabled':
      case 'enabled':
        startBleScan()
        break
      default:
        console.log('this should not happen!')
        break
    }
  }
```
So, in order to have a working code, I have to do this with a try and catch:
```javascript
  const activateLocation = async () => {
    try {
      const res: PromptFuncResponse = await RNAndroidLocationEnabler.promptForEnableLocationIfNeeded({
        interval: 10000,
        fastInterval: 5000
      })
      console.log('promptForEnableLocationIfNeeded', res)
      switch (res) {
        case 'already-enabled':
        case 'enabled':
          startBleScan()
          break
        default:
          console.log('this should not happen!')
          break
      }
    } catch (error) {
      console.warn('error when activating GPS', error.code)
    }
  }
```
And the following error:
```
WARN     error when activating GPS ERR00
```
As you can see, error is not typed, so I have to "guess" that it's the error from `ErrorCode` type.

It should be improved a little I think. I don't know if it's a good practice to return a conjunction like:
```javascript
const res: PromptFuncResponse | PromptErrorResponse = await RNAndroidLocationEnabler.promptForEnableLocationIfNeeded(...)
```

**Comment by Richou** [comment]
Well I think it is the normal behavior, If you refuse, the module consider that it is an error and "throws" an error with code ERR00 which means "The user has clicked on Cancel button in the popup".

If you want to get rid of the try catch in your activateLocation method, I can suggest you to wrap the call of RNAndroidLocationEnabler.promptForEnableLocationIfNeeded in other method like : 

```javascript
interface EnableResponse {
  message: string
}

interface EnableError {
  code: string;
  message: string;
}

function enableLocation(): Promise<EnableResponse | EnableError> {
    try {
      const res: PromptFuncResponse = await RNAndroidLocationEnabler.promptForEnableLocationIfNeeded({
        interval: 10000,
        fastInterval: 5000
      })
      return {
        message: 'OK',
      }
    } catch (error) {
      return error
    }
}
```

This way you can call this new method like this : 

```javascript
  const activateLocation = async () => {
      const res = await enableLocation({
        interval: 10000,
        fastInterval: 5000
      })
      if (!('error' in res)) {
        startBleScan()
      }
  }
```

Or something like this.
It is in fact what you suggested but managed in your side.

**Comment by MBach** [comment]
OK why not, but if someone wants to use `promptForEnableLocationIfNeeded` multiple times, he will have to write this managed code.
I'm currently working with the library [`react-native-ble-plx`](https://github.com/Polidea/react-native-ble-plx), and when a scan is starting, the following signature is:
https://polidea.github.io/react-native-ble-plx/#blemanagerstartdevicescan
```javascript
    startDeviceScan(
      UUIDs: UUID[] | null,
      options: ScanOptions | null,
      listener: (error: BleError | null, scannedDevice: Device | null) => void
    ): void
```

Somewhere else in your code, there is something like:
```javascript
  const onScanResult = (error: BleError | null, scannedDevice: Device | null) => {
    if (error) {
      console.warn(error)
    } else if (scannedDevice) {
      console.log(scannedDevice.id, scannedDevice.name)
    }
  }
```
So there's no try & catch, and no need to managed the code by yourself (I'm kind of lazy, like every programmer :)

**Comment by Richou** [comment]
Yeah the library you mentioned uses callbacks, I use promises.

**Pull request merged**